File: src/main/java/org/apache/fineract/client/services/ClientFamilyMemberApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface ClientFamilyMemberApi {
  /**
   * 
   * 
   * @param clientId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("clients/{clientId}/familymembers")
  Observable<String> addClientFamilyMembers(
    @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param familyMemberId  (required)
   * @param clientId clientId (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @DELETE("clients/{clientId}/familymembers/{familyMemberId}")
  Observable<String> deleteClientFamilyMembers(
    @retrofit2.http.Path("familyMemberId") Long familyMemberId, @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param familyMemberId  (required)
   * @param clientId clientId (required)
   * @return Observable&lt;String&gt;
   */
  @GET("clients/{clientId}/familymembers/{familyMemberId}")
  Observable<String> getFamilyMember(
    @retrofit2.http.Path("familyMemberId") Long familyMemberId, @retrofit2.http.Path("clientId") Long clientId
  );

  /**
   * 
   * 
   * @param clientId  (required)
   * @return Observable&lt;String&gt;
   */
  @GET("clients/{clientId}/familymembers")
  Observable<String> getFamilyMembers(
    @retrofit2.http.Path("clientId") Long clientId
  );

  /**
   * 
   * 
   * @param clientId  (required)
   * @return Observable&lt;String&gt;
   */
  @GET("clients/{clientId}/familymembers/template")
  Observable<String> getTemplate2(
    @retrofit2.http.Path("clientId") Long clientId
  );

  /**
   * 
   * 
   * @param familyMemberId  (required)
   * @param clientId clientId (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("clients/{clientId}/familymembers/{familyMemberId}")
  Observable<String> updateClientFamilyMembers(
    @retrofit2.http.Path("familyMemberId") Long familyMemberId, @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Body String body
  );

}

File: src/main/kotlin/org/openapitools/client/apis/ClientFamilyMemberApi.kt
package org.openapitools.client.apis

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path


interface ClientFamilyMemberApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/clients/{clientId}/familymembers")
    suspend fun addClientFamilyMembers(
        @Path("clientId") clientId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param familyMemberId
     * @param clientId clientId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @DELETE("v1/clients/{clientId}/familymembers/{familyMemberId}")
    suspend fun deleteClientFamilyMembers(
        @Path("familyMemberId") familyMemberId: Long,
        @Path("clientId") clientId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param familyMemberId
     * @param clientId clientId
     * @return [kotlin.String]
     */
    @GET("v1/clients/{clientId}/familymembers/{familyMemberId}")
    suspend fun getFamilyMember(
        @Path("familyMemberId") familyMemberId: Long,
        @Path("clientId") clientId: Long
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId
     * @return [kotlin.String]
     */
    @GET("v1/clients/{clientId}/familymembers")
    suspend fun getFamilyMembers(@Path("clientId") clientId: Long): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId
     * @return [kotlin.String]
     */
    @GET("v1/clients/{clientId}/familymembers/template")
    suspend fun getTemplate2(@Path("clientId") clientId: Long): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param familyMemberId
     * @param clientId clientId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/clients/{clientId}/familymembers/{familyMemberId}")
    suspend fun updateClientFamilyMembers(
        @Path("familyMemberId") familyMemberId: Long,
        @Path("clientId") clientId: Long,
        @Body body: String? = null
    ): String

}
